userguide/storage.rst

```diff
@@ -770,6 +781,47 @@ usually not worth the performance hit.
    dataset are saved, giving the effect of deduplication without the
    overhead.
 
+#ifdef freenas
+.. _Deduplication RAM requirements:
+
+Deduplication RAM requirements
+------------------------------
+
+A generally quoted rule of thumb is 5 GB of RAM per terabyte of 
+deduplicated storage. However, if the pool is highly duplicated 
+(around 3x to 5x space saving) this can drop to as low as 1 GB to 2 
+GB of RAM per TB of deduplicated storage. The exact size required for 
+deduplication data is found by using the :command:`CLI` command 
+:command:`zdb -U /data/zfs/zpool.cache -D pool_name` to identify the 
+total number of blocks in the pool, and the number of bytes of RAM 
+required per block (shown as *"{number} in core"* in the output). The 
+command  :command:`zdb -U /data/zfs/zpool.cache -S pool_name` 
+displays an estimate of the storage saving if deduplication is 
+applied to a dataset. Note that these commands may take a long time 
+to run because they scan the entire pool to produce the results. 
+
+There is also another RAM limitation affecting dedup.  Because ZFS 
+uses RAM for many caching activities, it limits the amount of metadata
+(including DDT data) for which it attempts to reserves space in RAM. By
+default, DDTs and other file system metadata can rely on up to 25% of
+the total RAM available to ZFS (known as ARC), and the balance is used 
+for  other caching. Therefore if the system has 32 GB of RAM of which 
+20 GB is available for ZFS, ZFS will aim to apportion this 25% (5 GB) 
+for metadata and 75% (15 GB) for file data. The DDTs can rely on not more 
+than 5 GB of RAM (and perhaps less), which may only allow dedup of 
+1 TB in many cases.
+
```

Comment:
Lots of asides here. Rework the sentences to get rid of those.

Reply:
Text has been updated - please re-review current version